query/aql_processor.go

```diff
@@ -1276,7 +1283,7 @@ func (qc *AQLQueryContext) FindDeviceForQuery(memStore memstore.MemStore, prefer
 
 func (qc *AQLQueryContext) runBatchExecutor(e BatchExecutor, isLastBatch bool) {
 	start := utils.Now()
-	e.preExec(isLastBatch, start)
+	e.preExec(isLastBatch, start, !qc.OOPK.currentBatch.indexVectorD.isNull())
```

Comment:
looks like param hasRegularFilter is redundent, we can just check indexVectorD.isNull() directly before we initialize indexVector

Reply:
true.. somehow I thought indexVectorD is not available from batchExecutor.